### PR TITLE
More JS store tests featuring fileUploadStore and a few simple ones

### PR DESCRIPTION
**Pull request by PVince81**
Brings in the following:
- fileUploadStore tests
- guestNameStore tests
- newGroupConversationStore tests
- quoteReplyStore tests
- sidebarStore tests

For the other stores not covered:
- they are too trivial (just getters/setters)
- they are blocked by global store usage (requireshttps://github.com/nextcloud/spreed/pull/5526)
- they are not fully testable because of the initial loadState / BrowserStore.getItem: we might want to move to a more explicit way of loading state with a "loadState" action of sorts => to be considered separately

~~Currently blocked because "temporaryMessage" is accessing the global store and I didn't find a way to set the current mocked store to the global module.~~

~~Maybe need to refactor those usages and always receive the store injected in the function.
At least that's the quickest workaround I can think of.~~

~~The long term approach would be to refactor all those usages and move them into the store as store actions where applicable (like `createTemporaryMessage`)~~

### CODE REVIEW

File: src/store/fileUploadStore.spec.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { createLocalVue } from '@vue/test-utils'
+import storeConfig from './storeConfig'
+import Vuex from 'vuex'
+import { cloneDeep } from 'lodash'
+
+import globalStore from '../store/index'
+
+// workaround for files using the store globally
+jest.mock('../store/index', () => jest.fn())
```
Comment (PVince81) [inline src/store/fileUploadStore.spec.js:9]: we can't put an object in here without redefining it completely, so the closest we can mock is a function, but the users of the module expect an object, not a function

**Comment by PVince81** [comment]
looks like there wouldn't be that much to adjust:
```
src/services/settingsService.js|26 col 23| import store from '../store/index'
src/services/conversationsService.js|27 col 23| import store from '../store/index'
src/services/messagesService.js|25 col 23| import store from '../store/index'
src/services/participantsService.js|36 col 23| import store from '../store/index'
src/utils/webrtc/models/LocalCallParticipantModel.js|22 col 29| import store from '../../../store/index.js'
src/utils/webrtc/models/LocalMediaModel.js|22 col 29| import store from '../../../store/index.js'
src/utils/webrtc/webrtc.js|31 col 26| import store from '../../store/index.js'
src/utils/fileUpload.js|23 col 23| import store from '../store/index'
src/utils/temporaryMessage.js|23 col 23| import store from '../store/index'
```

**Comment by PVince81** [comment]
raised https://github.com/nextcloud/spreed/pull/5524 and https://github.com/nextcloud/spreed/pull/5526 to fix store usage

**Comment by PVince81** [comment]
- [ ] separate the bug fix from fileUpload.js or is it trivial enough ?

**Comment by PVince81** [comment]
I'll handle participantsStore and messagesStore separately as they are more complex.

Please review the above, the only complex one is the file upload store.

File: .eslintrc.js [hunk 2]
```diff
@@ -2,4 +2,12 @@ module.exports = {
 	extends: [
 		'@nextcloud'
 	],
+	overrides: [
+    {
```
Comment (nickvergessen) [inline .eslintrc.js:6]: tabs vs. spaces in this file

File: jest.config.js [hunk 3]
```diff
@@ -26,7 +26,10 @@ module.exports = {
 	testMatch: ['<rootDir>/src/**/*.(spec|test).(ts|js)'],
 	resetMocks: false,
 	setupFiles: ['jest-localstorage-mock'],
-	setupFilesAfterEnv: ['<rootDir>/src/test-setup.js'],
+	setupFilesAfterEnv: [
+		'<rootDir>/src/test-setup.js',
+	    'jest-mock-console/dist/setupTestFramework.js',
```
Comment (nickvergessen) [inline jest.config.js:31]: tabs  vs. spaces

**Review by nickvergessen — APPROVED** [review]
Looks okay otherwise

**Comment by PVince81** [comment]
tabs fixed, waiting for CI now

**Pull request merged**